src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -190,23 +182,28 @@ def __call__(
         :param inp: The input to run the engine with. We expect a
             list of numpy arrays that contain the input ids,
             attention mask, and position ids (optionally)
+        :param kv_cache: The DecoderKVCache object that contains
+            the kv cache state
         :param val_inp: Whether the input is for validation or not
+
         :return: The generated token and corresponding logits
         """
         timer = self.timer_manager.current
-        if self.kv_cache:
+        if self.kv_cache_enabled:
```

Comment:
Hm, I feel the name `kv_cache_enabled` is a bit misleading now that the NLDecoderEngine no longer owns the kv_cache. Would it make sense to call it `kv_cache_required` and throw an error is the kv_cache passed in is None?

Reply:
good call!